source/nodejs/adaptivecards/src/utils.ts

```diff
@@ -25,21 +25,43 @@ export function appendChild(node: Node, child: Node) {
 }
 
 export function setProperty(target: any, propertyName: string, propertyValue: any, defaultValue: any = undefined) {
-	if (propertyValue && (!defaultValue || defaultValue !== propertyValue)) {
+    let targetIsAlreadySet = target[propertyName] !== undefined;
+
+    if (propertyValue === undefined || propertyValue === null) {
+        if (!targetIsAlreadySet) {
+            delete target[propertyName];
+        }
+    }
+    else if (propertyValue !== defaultValue) {
 		target[propertyName] = propertyValue;
 	}
 	else {
-		delete target[propertyName];
+        if (!targetIsAlreadySet) {
```

Comment:
nit: this statement can be merged with LN 30

Reply:
It can't, otherwise the else if that's in between wouldn't be hit. The sequential aspect is important here.